Source/BugsnagStackframe.m

```diff
@@ -71,17 +71,22 @@ - (NSDictionary *)toDictionary {
     BSGDictInsertIfNotNil(dict, self.method, BSGKeyMethod);
     BSGDictInsertIfNotNil(dict, self.machoUuid, BSGKeyMachoUUID);
 
-    NSString *frameAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, self.frameAddress];
-    BSGDictSetSafeObject(dict, frameAddr, BSGKeyFrameAddress);
-
-    NSString *symbolAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, self.symbolAddress];
-    BSGDictSetSafeObject(dict, symbolAddr, BSGKeySymbolAddr);
-
-    NSString *imageAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, self.machoLoadAddress];
-    BSGDictSetSafeObject(dict, imageAddr, BSGKeyMachoLoadAddr);
-
-    NSString *vmAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, self.machoVmAddress];
-    BSGDictSetSafeObject(dict, vmAddr, BSGKeyMachoVMAddress);
+    if (self.frameAddress) {
+        NSString *frameAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, [self.frameAddress unsignedLongValue]];
+        BSGDictSetSafeObject(dict, frameAddr, BSGKeyFrameAddress);
+    }
+    if (self.symbolAddress) {
+        NSString *symbolAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, [self.symbolAddress unsignedLongValue]];
+        BSGDictSetSafeObject(dict, symbolAddr, BSGKeySymbolAddr);
+    }
+    if (self.machoLoadAddress) {
+        NSString *imageAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, [self.machoLoadAddress unsignedLongValue]];
+        BSGDictSetSafeObject(dict, imageAddr, BSGKeyMachoLoadAddr);
+    }
+    if (self.machoVmAddress) {
+        NSString *vmAddr = [NSString stringWithFormat:BSGKeyFrameAddrFormat, [self.machoVmAddress unsignedLongValue]];
+        BSGDictSetSafeObject(dict, vmAddr, BSGKeyMachoVMAddress);
+    }
```

Comment:
Seems somewhat convoluted now to convert to long just so it can be formatted into a string - can we just get the string value and change the `BSGKeyFrameAddrFormat`?

Reply:
It seems a big long-winded but my understanding is that the API field relies on having hexadecimal addresses as a string. I'd be wary of changing the serialized response, unless I'm misunderstanding the review comment